### PR TITLE
[Layer] Add eval mode for the training

### LINKED ISSUE #847: Add training hint to the forwarding
for some layers, 

there should be 4 versions of forwarding. if current forwarding is inferencing is not known to the layer so it should be cued

|               | training | inferencing |
|---------------|----------|-------------|
| trainable     |              |             |
| non-trainable |          |             |

### PR DESCRIPTION
**Pull request by zhoonit**
**Changes proposed in this PR:**
- This patch add eval mode for the training forward and
fix batch normalization layer accordingly

**Self evaluation:**
1. Build test: [X]Passed [ ]Failed [ ]Skipped
2. Run test: [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: Jihoon Lee <jhoon.it.lee@samsung.com>

resolves #847

### CODE REVIEW

**Review by lhs8928 — APPROVED** [review]
LGTM

**Review by jijoongmoon — APPROVED** [review]
LGTM

**Comment by zhoonit** [comment]
Please wait for @kparichay 's approval before merge, there was an offline comment about behavior of bn_layer.h

**Comment by kparichay** [comment]
The two properties - `trainable` and `training` have different meaning and significance. I could explain but https://stackoverflow.com/a/50224517 already explains well. 
Tensorflow 2.0 merged the two terms for simplicity and state about this in - https://github.com/tensorflow/tensorflow/blob/095272a4dd259e8acd3bc18e9eb5225e7a4d7476/tensorflow/python/keras/layers/normalization_v2.py#L26
However, other frameworks like PyTorch and TF1.0 still allows both different configurations.
Let's keep both the terms separated and meaningful, and choose defaults that simulate TF2.0 behavior still providing the functionality of TF1.0 and PyTorch.

**Comment by kparichay** [comment]
After offline discussion, let's go forward with the current implementation.

**Review by kparichay — APPROVED** [review]
